mtags/src/main/scala-3/scala/meta/internal/pc/completions/MatchCaseCompletions.scala

```diff
@@ -341,6 +323,49 @@ object CaseKeywordCompletion:
         defnSymbols.getOrElse(semancticName, -1)
       }
 
+  def subclassesForType(tpe: Type)(using Context): List[Symbol] =
+    /**
+     * Split type made of & and | types to a list of simple types.
+     * For example, `(A | D) & (B & C)` returns `List(A, D, B, C).
+     * Later we use them to generate subclasses of each of these types.
+     */
+    def getParentTypes(tpe: Type, acc: List[Symbol]): List[Symbol] =
+      tpe match
+        case AndType(tp1, tp2) =>
+          getParentTypes(tp2, getParentTypes(tp1, acc))
+        case OrType(tp1, tp2) =>
+          getParentTypes(tp2, getParentTypes(tp1, acc))
+        case t =>
+          tpe.typeSymbol :: acc
+
+    /**
+     * Check if `sym` is a subclass of type `tpe`.
+     * For `class A extends B with C with D` we have to construct B & C & D type,
+     * because `A <:< (B & C) == false`.
+     */
+    def isExhaustiveMember(sym: Symbol): Boolean =
```

Comment:
I changed the approach , because it didn't work for types like `A | (B & C)` (as @tgodzik noticed). The issue was, that for `class A extends B,C,D`, `A <:< (E | (B & C)) == false`. Previously, I was collecting toplevel `&` types ( in this example, `E | (B & C)`) and trying to find parent (`B`, `C`,`D`) matching each of them. Now I reduce parent types to `B & C & D` type and test if the result is a subtype of `E | (B & C)`